src/braket/default_simulator/openqasm/_helpers/quantum.py

```diff
@@ -54,17 +54,16 @@ def is_controlled(phase: QuantumPhase) -> bool:
     return False
 
 
-def convert_phase_to_gate(controlled_phase: QuantumPhase) -> QuantumGate:
+def convert_phase_to_gate(controlled_phase: QuantumPhase) -> Union[QuantumGate, list[QuantumGate]]:
```

Comment:
This approach increases complexity, particularly in `inline_gate_def_body` which has a lot of logic embedded already. I wonder if we shouldn't be handling "controlled phase instructions" logic during interpretation, but rather in `ProgramContext.add_phase_instruction`. I guess it comes down to whether controlled phases are considered part of the spec, or up to implementation (which now we do a mix of both)